e2e/evm/util.go

```diff
@@ -0,0 +1,65 @@
+package evm
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/ChainSafe/chainbridge-core/chains/evm/calls/consts"
+	"github.com/ChainSafe/chainbridge-core/util"
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/rs/zerolog/log"
+)
+
+func WaitForProposalExecuted(client TestClient, bridge common.Address) {
+	startBlock, _ := client.LatestBlock()
+
+	query := ethereum.FilterQuery{
+		FromBlock: startBlock,
+		Addresses: []common.Address{bridge},
+		Topics: [][]common.Hash{
+			{util.ProposalEvent.GetTopic()},
+		},
+	}
+	ch := make(chan types.Log)
+
+	sub, err := client.SubscribeFilterLogs(context.Background(), query, ch)
+	if err != nil {
+		panic(err)
+	}
+	defer sub.Unsubscribe()
+
+	a, err := abi.JSON(strings.NewReader(consts.BridgeABI))
+	if err != nil {
+		panic(err)
```

Comment:
I think it would be better to return this error and then decide in setup itself what to do if the error occurs